diff --git a/include/smb2/libsmb2.h b/include/smb2/libsmb2.h
index f5f14d0..4255b7d 100644
--- a/include/smb2/libsmb2.h
+++ b/include/smb2/libsmb2.h
@@ -552,6 +552,7 @@ int smb2_get_nterror(struct smb2_context *smb2);
 struct smb2_url {
         const char *domain;
         const char *user;
+        const char *password;
         const char *server;
         const char *share;
         const char *path;
diff --git a/lib/init.c b/lib/init.c
index bd7b13e..ae52f42 100644
--- a/lib/init.c
+++ b/lib/init.c
@@ -227,6 +227,13 @@ struct smb2_url *smb2_parse_url(struct smb2_context *smb2, const char *url)
                 *(tmp++) = '\0';
                 u->user = strdup(ptr);
                 ptr = tmp;
+
+                if ((tmp = strchr(ptr, ':')) != NULL) {
+                        *(tmp++) = '\0';
+                        u->password = strdup(ptr);
+                        ptr = tmp;
+                }
+
         }
         /* server */
         if ((tmp = strchr(ptr, '/')) != NULL) {
@@ -259,6 +266,7 @@ void smb2_destroy_url(struct smb2_url *url)
         }
         free(discard_const(url->domain));
         free(discard_const(url->user));
+        free(discard_const(url->password));
         free(discard_const(url->server));
         free(discard_const(url->share));
         free(discard_const(url->path));
