diff --git a/include/smb2/libsmb2.h b/include/smb2/libsmb2.h
index f5f14d0..4255b7d 100644
--- a/include/smb2/libsmb2.h
+++ b/include/smb2/libsmb2.h
@@ -552,6 +552,7 @@ int smb2_get_nterror(struct smb2_context *smb2);
 struct smb2_url {
         const char *domain;
         const char *user;
+        const char *password;
         const char *server;
         const char *share;
         const char *path;
diff --git a/lib/init.c b/lib/init.c
index bd7b13e..102b636 100644
--- a/lib/init.c
+++ b/lib/init.c
@@ -225,8 +225,18 @@ struct smb2_url *smb2_parse_url(struct smb2_context *smb2, const char *url)
         /* user */
         if ((tmp = strchr(ptr, '@')) != NULL && strlen(tmp) > len_shared_folder) {
                 *(tmp++) = '\0';
-                u->user = strdup(ptr);
+
+                char *pwdtmp;
+                if ((pwdtmp = strchr(ptr, ':')) != NULL) {
+                        *(pwdtmp++) = '\0';
+                        u->user = strdup(ptr);
+                        u->password = strdup(pwdtmp);
+                } else {
+                        u->user = strdup(ptr);
+                }
+
                 ptr = tmp;
+
         }
         /* server */
         if ((tmp = strchr(ptr, '/')) != NULL) {
@@ -259,6 +269,7 @@ void smb2_destroy_url(struct smb2_url *url)
         }
         free(discard_const(url->domain));
         free(discard_const(url->user));
+        free(discard_const(url->password));
         free(discard_const(url->server));
         free(discard_const(url->share));
         free(discard_const(url->path));
