diff --git a/cmake/ConfigureChecks.cmake b/cmake/ConfigureChecks.cmake
index a28846a..1340f1d 100644
--- a/cmake/ConfigureChecks.cmake
+++ b/cmake/ConfigureChecks.cmake
@@ -7,7 +7,9 @@ check_include_file("arpa/inet.h" HAVE_ARPA_INET_H)
 check_include_file("dlfcn.h" HAVE_DLFCN_H)
 check_include_file("fcntl.h" HAVE_FCNTL_H)
 check_include_file("gssapi/gssapi.h" HAVE_GSSAPI_GSSAPI_H)
-check_include_file("krb5/krb5.h" HAVE_LIBKRB5)
+if(LibKrb5_FOUND)
+  add_definitions(-DHAVE_LIBKRB5)
+endif()
 check_include_file("inttypes.h" HAVE_INTTYPES_H)
 check_include_file("netdb.h" HAVE_NETDB_H)
 check_include_file("netinet/in.h" HAVE_NETINET_IN_H)
